1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [43]:
import numpy as np
import pandas as pd

In [ ]:
covXY=M[(X−M(X))(Y−M(Y))]=M(XY)−M(X)M(Y)

In [33]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])

In [34]:
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [35]:
covZpKs = np.mean(zp*ks) - np.mean(zp)*np.mean(ks)
covZpKs

9157.839999999997

In [37]:
covZK = np.cov(zp, ks, ddof =0)
covZK

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Коэффициент Пирсона

In [39]:
σzp = np.std(zp)
σzp

59.115480206118605

In [40]:
σks = np.std(ks)
σks

174.55340157098058

In [41]:
r = covZpKs /(σzp*σks)
r

0.8874900920739158

In [42]:
r = np.corrcoef(zp, ks)
r

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [49]:
df = pd.DataFrame({'zp': [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], 'ks': [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]})
df

zp   ks
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [50]:
df.corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

2.Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Так как неизвестно среднее квадратичное отклонение генеральной совокупности, используем критерий Стьюдента.
X ± tα/2 * σ/√n-1

In [52]:
X = np.mean([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
X

118.1

In [54]:
t = 2.262

In [57]:
σ = np.std([131, 125, 115, 122, 131, 115, 107, 99, 125, 111], ddof =1)
σ

10.54566788359614

In [58]:
LX = X-t*σ/pow(9, 0.5)
LX

110.1485664157685

In [60]:
UX = X+t*σ/pow(9, 0.5)
UX

126.05143358423149

доверительный интервал для математического ожидания генеральной совокупности с надежностью 0.95 [110.1486 - 126.0514]

3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Так как известно среднее квадратичное отклонение генеральной совокупности, используем критерий Z.
X ± Zα/2 * σ/√n

In [63]:
σ = pow(25, 0.5)
σ

5.0

α = 5, a/2 = 2.5;
Z = 1.96
X = 174.2;
n = 27

In [64]:
LX = 174.2 - 1.96*5/pow(27, 0.5)
LX

172.31398912064722

In [66]:
UX = 174.2 + 1.96*5/pow(27, 0.5)
UX

176.08601087935276

доверительный интервал для математического ожидания генеральной совокупности с надежностью 0.95 [172.3140 - 176.0860]